### ISSUE TITLE
File limits are too restrictive

### ISSUE THREAD

**Issue by greg-hellings** [karthicraghupathi/pipenv-requirements-pre-commit#1]
Limits itself to when `Pipfile(\.lock)?` changes. It should also run when `requirements(-dev)?\.txt` changes.

**Comment by karthicraghupathi** [comment]
The requirement files are produced by running `pipenv requirements`. All changes should be done only using `pipenv` which will modify the `Pipfile` and consequently produces the lock file.

**Comment by greg-hellings** [comment]
Yes. And if this runs on changes to requirements files it can catch when someone alters those without invoking pipenv, in addition to its current function of updating them when Pipfile changes

**Comment by karthicraghupathi** [comment]
If the `Pipfile` has not been updated, then `pipenv requirements` will have no knowledge of what changed. Adding to the requirements file directly is not the prescribed way of doing things when the project has been setup with `pipenv` to manage dependencies.

**Comment by greg-hellings** [comment]
I understand that, and I don't disagree. But as it currently stands, if a user modifies the requirements files there is no way to detect that. This hook should be able to do that detection just by running when there are changes in one of the requirements files. No? That way it can programmatically detect changes to requirements.txt during a lint step in CI and prevent users from not inserting those values into Pipfiles?

**Comment by karthicraghupathi** [comment]
While I would like for this hook to catch modifications made directly to the requirements file, I don't think there is a good way to accomplish this.

Consider this scenario: a developer `pip installs some-package` and then runs `pip freeze -r > requirements.txt`.

Now the requirement file has changed. When `some-package` was installed, it bought in a bunch of other dependencies. How would we now determine which was the installed package and which package are dependencies of `some-package` to update the `Pipfile`.

IMHO this is over complicating the whole thing. Sticking to the prescribed way of doing this is the simplest and best approach.

That said, I could envision a scenario where the current requirements file is compared with the output file produced by this hook and fail the pre-commit step if the files don't match. This will at least prevent users from modifying the requirements file directly. But if this hook is part of the CI step, then that shouldn't matter as the requirements file will always get overwritten.
